opentracing-impl/src/main/java/io/opentracing/impl/AbstractTracer.java

```diff
@@ -35,6 +34,8 @@ protected AbstractTracer() {
     }
 
     abstract AbstractSpanBuilder createSpanBuilder(String operationName);
+    
+    abstract AbstractSpanContext createSpanContext(Map<String, Object> traceState);
```

Comment:
If I were designing this abstract implementation, I would've done it as:
```java
abstract class AbstractTracer<SPAN extends AbstractSpan, SCTX extends AbstractSpanContext> ... {
    abstract SPAN createSpanContext(Map<String, Object> traceState);
}

The implementations then would deal with their own concrete types instead of abstract superclass, and the abstract tracer would take care of the necessary downcasting (as long as it gets a concrete Class in the constructor)
```